src/pudl/analysis/pudl_models.py

```diff
@@ -48,7 +51,193 @@ def core_sec10k__quarterly_company_information() -> pd.DataFrame:
+    df["zip_code_4"].where(df["zip_code"].str.len() > 5, None)
+    df["name_change_date"] = pd.to_datetime(df["name_change_date"], format="%Y%m%d")
+    df["state"] = df["state"].str.upper()
+    df["state_of_incorporation"] = df["state_of_incorporation"].str.upper()
+    df[["industry_name_sic", "industry_id_sic"]] = df["industry_id_sic"].str.extract(
+        r"^(.+)\[(\d{4})\]$"
+    )
+    df["sec_act"] = df["sec_act"].where(df["sec_act"].isnull(), "1934 act")
+    # fiscal year end should conform to MMDD format
+    df["fiscal_year_end"] = df["fiscal_year_end"].str.zfill(4)
+    df.loc[
+        ~df.fiscal_year_end.str.contains(
+            r"^(?:(?:0[1-9]|1[0-2])(?:0[1-9]|1\d|2\d|3[01])|(?:0[13-9]|1[0-2])(?:29|30)|(?:0[13578]|1[02])31)$",
+            na=False,
+        ),
+        "fiscal_year_end",
+    ] = None
+    # make taxpayer ID a 9 digit number with a dash separating the first two digits
+    df["taxpayer_id_irs"] = df["taxpayer_id_irs"].str.replace("-", "", regex=False)
+    df["taxpayer_id_irs"] = df["taxpayer_id_irs"].where(
+        (df["taxpayer_id_irs"].str.len() == 9)
+        & (df["taxpayer_id_irs"].str.isnumeric().all()),
+        pd.NA,
+    )
+    df["taxpayer_id_irs"] = (
+        df["taxpayer_id_irs"].str[:2] + "-" + df["taxpayer_id_irs"].str[-7:]
+    )
+    df = convert_cols_dtypes(df, data_source="sec10k")
+    df = df.sort_values(by=["central_index_key", "report_date"])
+
+    return df
+
+
+@asset(
+    io_manager_key="pudl_io_manager",
+    group_name="sec10k",
+)
+def out_sec10k__quarterly_company_information(
+    core_sec10k__quarterly_company_information: pd.DataFrame,
+    core_eia__entity_utilities: pd.DataFrame,
+) -> pd.DataFrame:
+    """Company information extracted from SEC10k filings and matched to EIA utilities."""
+    matched_df = _load_table_from_gcs("out_sec10k__parents_and_subsidiaries")
+    matched_df = (
+        matched_df[["central_index_key", "utility_id_eia"]]
+        .dropna()
+        .drop_duplicates(
+            subset="central_index_key"
+        )  # matches should already be 1-to-1 but drop duplicates to ensure this is true
+    )
+    out_df = core_sec10k__quarterly_company_information.merge(
+        matched_df, how="left", on="central_index_key"
+    )
+    # merge utility name on
+    out_df = out_df.merge(core_eia__entity_utilities, how="left", on="utility_id_eia")
+    return out_df
+
+
+@asset(io_manager_key="pudl_io_manager", group_name="sec10k")
+def core_sec10k__assn__sec10k_filers_and_eia_utilities() -> pd.DataFrame:
+    """An association table between SEC 10k filing companies and EIA utilities."""
+    matched_df = _load_table_from_gcs("out_sec10k__parents_and_subsidiaries")
+    matched_df = (
+        matched_df[["central_index_key", "utility_id_eia"]]
+        .dropna()
+        .drop_duplicates(
+            subset="central_index_key"
+        )  # matches should already be 1-to-1 but drop duplicates to ensure this is true
+    )
+    return matched_df
```

Comment:
Instead of duplicating this code from the `out_sec10k__quarterly_company_information` table should I make the `out_sec10k__quarterly_company_information` asset a multi asset which creates two outputs - the association table and the output table?